tests/sim/fungible_proxy.rs

```diff
@@ -0,0 +1,278 @@
+        fungible_request_proxy.ft_on_transfer(alice.account_id(), send_amt.0.to_string(), args.into())
+    );
+    assert_one_promise_error(result, "amount smaller than fee_amount")
+}
+
+#[test]
+fn test_transfer_receiver_send_failed() {
+    let (alice, bob, builder, fungible_request_proxy, ft_contract) = init_fungible();
+
+    let send_amt = U128::from(500000000); // 500 USDC.e
+    let (alice_balance_before, _, _) =
+        fungible_transfer_setup(&alice, &bob, &builder, &ft_contract, send_amt);
+
+    // Previous line registers all accounts, so we unregister bob here
+    // Receiver is not registered with the token contract, so sending to it will fail
+    call!(bob, ft_contract.unregister_account(bob.account_id()));
+
+    // Constuct the `msg` argument using our contract
+    // Transferring 100 USD worth of USDC.e from alice to bob, with a 2 USD fee to builder
+    let args = PaymentArgs {
+            fee_address: "builder".try_into().unwrap(),
+            fee_amount: 200.into(), // 2 USD
+            payment_reference: "abc7c8bb1234fd12".into(),
+            to: bob.account_id().try_into().unwrap()
+    };
+
+    let result = call!(
+        ft_contract.user_account,
+        fungible_request_proxy.ft_on_transfer(alice.account_id(), send_amt.0.to_string(), args.into())
+    );
+    result.assert_success();
+    let change = result.unwrap_json::<String>().parse::<u128>().unwrap();
+
+    let alice_balance_after = call!(alice, ft_contract.ft_balance_of(alice.account_id()))
+        .unwrap_json::<U128>()
+        .0
+        + change;
+
+    // Ensure no balance changes / all funds returned to sender
+    assert!(alice_balance_after == alice_balance_before);
+}
+
+#[test]
+fn test_transfer_fee_receiver_send_failed() {
+    let (alice, bob, builder, fungible_request_proxy, ft_contract) = init_fungible();
+
+    let send_amt = U128::from(500000000); // 500 USDC.e
+    let (alice_balance_before, _, _) =
+        fungible_transfer_setup(&alice, &bob, &builder, &ft_contract, send_amt);
+
+    // Previous line registers all accounts, so we unregister builder here
+    // Fee_receiver is not registered with the token contract, so sending to it will fail
+    call!(
+        builder,
+        ft_contract.unregister_account(builder.account_id())
+    );
+
+    let args = PaymentArgs {
+            fee_address: "builder".try_into().unwrap(),
+            fee_amount: 200.into(),
+            payment_reference: "abc7c8bb1234fd12".into(),
+            to: bob.account_id().try_into().unwrap()
+    };
+
+    let result = call!(
+        ft_contract.user_account,
+        fungible_request_proxy.ft_on_transfer(alice.account_id(), send_amt.0.to_string(), args.into())
+    );
+    result.assert_success();
+    // The mocked fungible token does not handle change
+    let change = result.unwrap_json::<String>().parse::<u128>().unwrap();
+
+    assert_unchanged_balance(alice, alice_balance_before.sub(change), &ft_contract, "Alice");
```

Comment:
Consider also checking that the appropriate log message is emitted:

https://github.com/RequestNetwork/near-contracts/blob/8649fd6c7e010e8af8b498d3d732b2b1bd1b0939/fungible_proxy/src/lib.rs

```rust
                format!(
                    "Failed to transfer to account {}. Returning attached amount of {} of token {} to {}",
                    args.to, amount.0, token_address, payer)
                .as_bytes(),
```